src/index.js

```diff
@@ -152,3 +274,32 @@ export function denormalize(obj, entities, schema, bag = {}) {
     return denormalizeObject(entity, entities, schema, bag);
   }
 }
+
+/**
+ * Same as denormalize right above, but memoized
+ */
+function denormalizeMemoized(obj, entities, schema, bag = {}) {
```

Comment:
If we add an options argument here, we could skip the various `denormalize*Memoized()` functions and merge their code into the existing `denormalize*()`:

```js
export function denormalize(obj, entities, schema, options, bag = {}) {
  if (obj === null || typeof obj === 'undefined' || !isObject(schema)) {
    return obj;
  }

  if (schema instanceof EntitySchema) {
    return denormalizeEntity(obj, entities, schema, options, bag);
  } else if (schema instanceof IterableSchema) {
    return denormalizeIterable(obj, entities, schema, options, bag);
  } else if (schema instanceof UnionSchema) {
    return denormalizeUnion(obj, entities, schema, options, bag);
  }
  // Ensure we don't mutate it non-immutable objects
  const entity = isImmutable(obj) ? obj : merge({}, obj);
  return denormalizeObject(entity, entities, schema, options, bag);
}

```